### PR TITLE
Basic stock list written with React

### PR DESCRIPTION
**Pull request by liza8bit**
@ericyliu @dougvk @dminnear 

New items added via manually editing the JSON file show up just fine, but it seems that using the form on the site doesn't write correctly to the file (or something). Also, more minor, but I'm not sure how to get the value of the checkbox- from what I've found, the value is always the same whether or not the checkbox is actually checked.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+var fs = require('fs');
+var path = require('path');
+var express = require('express');
+var bodyParser = require('body-parser');
```
Comment (ericyliu) [inline server.js:16]: you might already know this, but generally for node projects, you want a package.json file in the root folder for local node_modules

http://stackoverflow.com/questions/12127114/how-to-write-a-package-json-file-so-that-all-dependencies-are-downloaded-with-n

that way, other people using your project can run 'npm install' and install all the libraries your project is dependent on

Comment (ericyliu) [inline server.js:16]: also once you do have a package.json file, you can install node modules by doing 'npm install node-module --save' and it'll automatically put a line in your package.json file under dependencies for that node-module

File: public/scripts/tutorial.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    this.props.onSubmit({category: cat, price: price, stocked: stock, name: name});
+    React.findDOMNode(this.refs.category).value='';
+    React.findDOMNode(this.refs.name).value='';
+    React.findDOMNode(this.refs.price).value='';
+  },
+  render: function() {
+    return (
+        <div className="addProduct">
+        <h3> Add a product </h3>
+        <form className="commentForm">
```
Comment (ericyliu) [inline public/scripts/tutorial.js]: you need a `onSubmit={this.handleSubmit}` attribute here, or else your handleSubmit function will never run

https://facebook.github.io/react/docs/tutorial.html#adding-new-comments

File: public/scripts/tutorial.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
+var AddProduct = React.createClass({
+  handleSubmit:function(e) {
+    e.preventDefault();
+    var cat=React.findDOMNode(this.refs.category).value.trim();
+    var name=React.findDOMNode(this.refs.name).value.trim();
+    var price = React.findDOMNode(this.refs.price).value.trim();
+    var stock = React.findDOMNode(this.refs.stocked).value();
```
Comment (ericyliu) [inline public/scripts/tutorial.js]: use the .checked property to see if a checkbox is checked

**Comment by ericyliu** [comment]
It's a really good start, you're almost to a working app (there's a few bugs i'll let you find)

at some point in your learning process, I would convert this to coffeescript since that's what you'll be using on the RMS project and the syntax is pretty different

File: public/scripts/tutorial.js (hunk 2 above)
Comment (dougvk) [inline public/scripts/tutorial.js]: ah yes. also, if you're wondering why something isn't working you should put `debugger;` in your code and open up the chrome developer tools when you run your application. this gives you an interactive debugger (as long as that line of code gets executed

Comment (dougvk) [inline public/scripts/tutorial.js]: @ericyliu I think @liza8bit is passing onSubmit a [different function](https://github.com/liza8bit/stocklist/pull/1/files#diff-d56dcef1dcd3108eb64737d609d57820R93). @liza8bit maybe try putting a `debugger;` statement [here](https://github.com/liza8bit/stocklist/pull/1/files#diff-d56dcef1dcd3108eb64737d609d57820R95) to figure out why it's not getting called

Comment (liza8bit) [inline public/scripts/tutorial.js]: i think @ericyliu was right; the form itself didn't have an onSubmit to
call, even though the component had a property onSubmit that was passed in.
I added it in and basic stuff is working now

Comment (ericyliu) [inline public/scripts/tutorial.js]: @dougvk I know handleAddProduct gets called after handleSubmit, but now that you mention it, I dont see what is calling handleAddProduct haha

Comment (dougvk) [inline public/scripts/tutorial.js]: correct! my bad. polymorphism problems.

Comment (ericyliu) [inline public/scripts/tutorial.js]: ohhh nevermind I see whats going on, she was passing the onSubmit function in ProductList

File: public/scripts/tutorial.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  },
+  componentDidMount: function(){
+    this.loadProducts();
+    setInterval(this.loadProducts, this.props.pollInterval);
+  },
+  render: function() {
+    return(
+        <div className="ProductList">
+        <h2>Products</h2>
+        <AddProduct onSubmit={this.handleAddProduct} />
```
Comment (ericyliu) [inline public/scripts/tutorial.js]: you have this onsubmit here, both get called

File: public/scripts/tutorial.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+  handleSubmit:function(e) {
+    e.preventDefault();
+    var cat=React.findDOMNode(this.refs.category).value.trim();
+    var name=React.findDOMNode(this.refs.name).value.trim();
+    var price = React.findDOMNode(this.refs.price).value.trim();
+    var stock = React.findDOMNode(this.refs.stocked).checked;
+    if(!cat || !name || !price){
+      return;
+    }
+    this.props.onSubmit({category: cat, price: price, stocked: stock, name: name});
```
Comment (ericyliu) [inline public/scripts/tutorial.js]: this then calls the other submit function `handleAddProduct`

**Comment by dougvk** [comment]
looks good to me (LGTM) - okay to merge. can you open up a new PR where you convert all the javascript to coffeescript? you can set coffee to [watch and compile the files to JS when they're saved](https://stackoverflow.com/questions/6286366/compile-coffeescript-on-save)

**Comment by dougvk** [comment]
@liza8bit when you merge, can you do a [squash merge](https://stackoverflow.com/questions/5308816/how-to-use-git-merge-squash) on master with a [nice commit message](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)? you could use [Git Town](https://github.com/Originate/git-town) too. up to you

**Comment by ericyliu** [comment]
hahahaha woops, oh well, do it for the next PR, also remember to delete your branch when youre done with it

**Comment by liza8bit** [comment]
@dougvk whoops, saw it a little late, I'll keep that in mind for next time

**Comment by dougvk** [comment]
:'-[

**Pull request merged**